### PR TITLE
[12.0] Add module stock_storage_type_putaway_strategy - alpha

### PR DESCRIPTION
**Pull request by grindtildeath**
Replaces: #5 
Depends on: #7

### CODE REVIEW

File: stock_location_storage_type_strategy/models/stock_location.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                # it on its children
+                if current_location.pack_storage_strategy != 'none':
+                    display = False
+                    break
+                current_location = current_location.location_id
+            location.display_pack_storage_strategy = display
+
+    def get_putaway_strategy(self, product):
+        putaway_location = super().get_putaway_strategy(product)
+        quant = self.env.context.get('storage_quant')
```
Comment (guewen) [inline stock_location_storage_type_strategy/models/stock_location.py]:
```suggestion
        quant = self.env.context.get('storage_quant')
        return self.get_storage_type_strategy(product, putaway_location, quant)
        
    def get_storage_type_strategy(self, product, putaway_location, quant=False)
```

Can we do something like this? It helps for inheritance, if we have to insert something between the put-away strategy and the storage strategy.

**Review by guewen — APPROVED** [review]
LG globally :)

File: stock_location_storage_type_strategy/models/stock_location.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+    _inherit = 'stock.location'
+
+    pack_storage_strategy = fields.Selection(
+        selection=[('none', 'None'),
+                   ('ordered_locations', 'Ordered Children Locations')],
+        required=True,
+        default='none',
+        string='Packs storage strategy',
+        help='TODO',
```
Comment (Tonow-c2c) [inline stock_location_storage_type_strategy/models/stock_location.py]: :fire:

File: stock_location_storage_type_strategy/tests/test_storage_type_strategy.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        #     'stock_putaway_storage_type_strategy.stock_putaway_rule_product_9_stock_cardboxes'
+        # )
+        # cls.product_lot_pallets_putaway = cls.product_pallets_putaway.copy(
+        #     {'product_id': cls.product_lot.id}
+        # )
+        # cls.product_lot_cardboxes_putaway = cls.product_cardboxes_putaway.copy(
+        #     {'product_id': cls.product_lot.id}
+        # )
+        # pick_types = cls.cls.receipts_picking_type | cls.internal_picking_type
+        # pick_types.write({'show_entire_packs': True})
```
Comment (Tonow-c2c) [inline stock_location_storage_type_strategy/tests/test_storage_type_strategy.py]: ?

**Comment by grindtildeath** [comment]
TODO: Add a note in usage for best practice. (e.g. Define a `stock.package.storage.location` without constraints to accept anything)

**Review by damdam-s — APPROVED** [review]
LGTM
@guewen is right about inheritance facility

**Comment by jgrandguillaume** [comment]
Migrated to v 13, so I close this one

**Pull request closed without merging**